File: LeetCode/Python/#504 Base 7.py
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return '0'
        n = abs(num)
        res = ""
        while n:
            res =str(n%7)+res
            n = n//7
        if num<0:
            return "-"+res
        return res



class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num ==0:
            return 0
        if num<0:
            return "-"+self.convertToBase7(-num)
        res = ''
        while num>0:
            res = str(num%7)+res
            num /= 7
        return res

File: LeetCode/JavaScript/#504 Base 7.js
/**
 * @param {number} num
 * @return {string}
 */
var convertToBase7 = function(num) {
    var res = [];
    var s;
    if(num<0){
      num = num*-1;
      res.push("-");
    }else{
      res.push("");
    } 
	while(num>0){
	  res.push(""+num%7);
	  num = Math.floor(num/7);
	}
	s = res[0]+res.slice(1).reverse().join("");
	return s;
};


// parseInt(num).toString(base) 